### DISCUSSION TITLE
In laravel mix, how do I reuse jquery import without creating multiple big js files?

### DISCUSSION THREAD

**Discussion by ProteanCode** [laravel/framework#36128] [Q&A]
I have an app.js file that imports bootstraps (require('bootstrap')), and this is fine. I use this file in my layout, but I need to use jQuery in one another page.

If I want to create a registration.js file where I do import jQuery (import * as $ from "jquery") I noticed that the output file is quite big. I want to reuse the jquery from previously required bootstrap.js

**Comment by ProteanCode** [comment]
Okay so I see that jquery got registered in window object, so can I depend on the order of chain calls in webpack.mix.js to know if jQuery is present at specific file further down in compilation process?

**Comment by veelasky** [comment]
try this:

```js
mix.webpackConfig({
    'externals': {
        jquery: 'jQuery'
    }
});
```

reference: 
- [webpack](https://webpack.js.org/configuration/externals/)
- [mix](https://laravel-mix.com/docs/6.0/quick-webpack-configuration)
